### PR TITLE
Preliminary implementation of complex arrays functions

### PR DESCRIPTION
**Pull request by facorazza**
It might better to move all these functions to a separate file like `sequtils.nim`. I'd also like to create tests for these functions but I'm having a hard time understanding the structure of the test directory.

### CODE REVIEW

**Comment by konsumlamm** [comment]
I don't see a good reason for adding these functions, vector math operations are definitely out of scope for `std/complex`. I'd recommend making a 3rd party library for this or using one of the existing vector math libraries.

File: lib/pure/complex.nim [hunk 1]
```diff
@@ -402,5 +402,139 @@ func `$`*(z: Complex): string =
     doAssert $complex(1.0, 2.0) == "(1.0, 2.0)"
 
   result = "(" & $z.re & ", " & $z.im & ")"
+  
+  func `+`*[T](x, y: openArray[Complex[T]]): seq[Complex[T]] =
+  ## Element-wise addition of two sequences of complex numbers.
+  assert x.len == y.len
```
Comment (juancarlospaco) [inline lib/pure/complex.nim:408]: Indentation error ?.

**Comment by beef331** [comment]
Based off the title I was half expecting the following which makes more sense for stdlib library to me:
```nim
proc `&`*[IdxA, IdxB: static int, T](a: array[IdxA, T], b: array[IdxB, T]): array[IdxB + IdxA, T] =
  for i, x in a:
    result[i] = x
  for i, x in b:
    result[i + IdxA] = x
    
proc `&`*[Idx: static int, T](val: T, arr: array[Idx, T]): array[Idx + 1, T] =
  result[0] = val
  for i, x in arr:
    result[i + 1] = x
    
proc `&`*[Idx: static int, T](arr: array[Idx, T], val: T): array[Idx + 1, T] =
  for i, x in arr:
    result[i] = x
  result[^1] = val
  
var 
  a = 10
  b = [10, 20] & a
  c = b & [10, 20, 30]
  d = a & c

assert b == [10, 20, 10]
assert c == [10, 20, 10, 10, 20, 30]
assert d == [10, 10, 20, 10, 10, 20, 30]
```

**Comment by Araq** [comment]
Sorry, rejected. Needs to start its life as a Nimble package.

**Pull request closed without merging**